src/MongoDB/Cursor.c

```diff
@@ -72,17 +72,32 @@ PHP_METHOD(Cursor, setTypeMap)
 
 static int php_phongo_cursor_to_array_apply(zend_object_iterator *iter, void *puser TSRMLS_DC) /* {{{ */
 {
-	zval **data, *return_value = (zval*)puser;
+#if PHP_VERSION_ID >= 70000
+	zval *data, *return_value = (zval*)puser;
+
+        data = iter->funcs->get_current_data(iter);
+#else
+        zval **data, *return_value = (zval*)puser;
 
 	iter->funcs->get_current_data(iter, &data TSRMLS_CC);
+#endif
 	if (EG(exception)) {
 		return ZEND_HASH_APPLY_STOP;
 	}
+#if PHP_VERSION_ID >= 70000
+        if (data == NULL) {
+#else
 	if (data == NULL || *data == NULL) {
+#endif
 		return ZEND_HASH_APPLY_STOP;
 	}
+#if PHP_VERSION_ID >= 70000
+        Z_ADDREF_P(data);
```

Comment:
Shouldn't this be Z_TRY_ADDREF_P() ? Are we sure its guaranteed to need a addref ?